src/RNGeth.ts

```diff
@@ -0,0 +1,112 @@
+import { NativeModules } from 'react-native'
+import { NodeConfig, Account } from './types'
+import { GethNativeModule } from "./GethNativeModule";
+
+export class RNGeth {
+  geth: GethNativeModule = NativeModules.RNGeth
+
+  /**
+   * Configure and prepare the node
+   * @returns success status of operation
+   */
+  async setConfig(config: NodeConfig) {
+    return this.geth.setConfig(config)
+  }
+
+  /**
+   * Start creates a live P2P node and starts running it.
+   * @returns success status of operation
+   */
+  async start(): Promise<boolean> {
+    return await this.geth.startNode()
+  }
+
+  /**
+   * Terminates a running node along with all it's services.
+   * @returns success status of operation
+   */
+  async stop(): Promise<boolean> {
+    return await this.geth.stopNode()
+  }
+
+  /**
+   * Subscribes to notifications about the current blockchain head
+   * @return true if subscribed
+   */
+  async subscribeNewHead(): Promise<boolean> {
+    return await this.geth.subscribeNewHead()
+  }
+
+  /**
+   * Add a new account
+   * @param privateKey - the hex-encoded private key
+   * @param passphrase - the passphrase used for the account
+   * @returns the new account
+   */
+  async addAccount(privateKey:string , passphrase: string): Promise<string> {
+    return await this.geth.addAccount(privateKey, passphrase)
+  }
+
+  /**
+   * Returns all key files present in the directory.
+   * @returns all accounts
+   */
+  async listAccounts(): Promise<Account[]> {
+    return await this.geth.listAccounts()
+  }
+
+  /**
+   * Unlock an account
+   * @param address - the address to unlock
+   * @param passphrase - the passphrase of the account
+   * @param timeout - unlock duration in seconds
+   * @returns the unlocked status of the account
+   */
+  async unlockAccount(account: string, passphrase: string, timeout: number): Promise<boolean> {
+    const oneSecond = 1000000000 // in Go, time.Second
+    return await this.geth.unlockAccount(account, passphrase, timeout * oneSecond)
```

Comment:
is this the right transformation based on the usage?

Reply:
The RPC unlockAccount is specified in seconds as far as I could see.
On the mobile side we use: 
```
export const UNLOCK_DURATION = 300
```
Which is actually also the default (300s).

On the go side, the `unlockAccount` method receives a `time.Duration` which is just a nominal integer. So in Go if I wanted a 300s duration I would do `d := 300 * time.Second` and `time.Second` equals `1000000000`.

Reply:
I'd be in favor of fixing things on the mobile side so we don't have to do any transformation here. The need for this transformation may cause confusion in the long run